lib/game/game_widget.dart

```diff
@@ -102,14 +113,30 @@ class _GameWidgetState extends State<GameWidget> {
   @override
   void initState() {
     super.initState();
+
+    // Add the initial overlays
+    _initActiveOverlays();
+
     addOverlaysListener(widget.game);
   }
 
+  void _initActiveOverlays() {
+    if (widget.activeOverlays != null) {
+      _checkOverlays(widget.activeOverlays.toSet());
+      widget.activeOverlays?.forEach((key) {
+        widget.game.overlays.add(key);
+      });
+    }
+  }
+
   @override
-  void didUpdateWidget(covariant GameWidget<Game> oldWidget) {
+  void didUpdateWidget(covariant GameWidget<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
     if (oldWidget.game != widget.game) {
       removeOverlaysListener(oldWidget.game);
+
+      // Reset the overlays
+      _initActiveOverlays();
```

Comment:
If the user changes the list passed as `activeOverlays`, the widget won't react to this change. Is that intended?

Reply:
I didn't thought on enabling an external control, I will think if this will generate any issues. Maybe we could add another block here to handle that, I will do some tests.